src/include/cpt/Optional.hpp

```diff
@@ -0,0 +1,41 @@
+//
+// Purpur Tentakel
+// cpp 23
+// 12.05.24
+//
+
+#pragma once
+#include <functional>
+#include <stdexcept>
+#include <variant>
+
+namespace cpt {
+    template<typename T>
+    class Otional final {
+    private:
+        std::variant<T, std::monostate> m_value;
+
+    public:
+        Otional(T value) : m_value{ std::in_place_index<0>, std::forward<T>(value) } { }
```

Comment:
where do i need the `T&&` ? 
like this?:
`Otional(T&& value) : m_value{ std::in_place_index<0>, std::forward<T>(value) } { }`

but how do I construct this with an lvalue then?
or does it always cast du an rvalue when I pass an lvalue into a function?

Reply:
Yes, like that.

If you use `&&` on a deduces type (like `T` in this case), this is ***not*** an rvalue reference, but a "forwarding reference" (or "universal reference") instead. This means that the type of `value` will be deduced to whatever was passed in (could be either lvalue, lvalue reference, lvalue const-reference, or rvalue [I hope I don't miss something here]).

```cpp
void f(auto&& param) // universal reference, because f is a template and the type of param is deduced

void f(std::string&& param) // rvalue reference
```
As you can see: Same syntax, different meaning.